Added the supported API classes (#109)
eclipse/microprofile-telemetry: tracing/spec/src/main/asciidoc/microprofile-telemetry-tracing-spec.asciidoc

To obtain the `Tracer` with the OpenTelemetry API, the consumer must use the exact same instrumentation name and version used by the implementation.
Failure to do so, may result in a different `Tracer` and incorrect handling of the OpenTelemetry data.

> === OpenTelemetry API Classes supported by this specification

Reviewer: ```suggestion
=== Supported OpenTelemetry API Classes
```

Classes from the following API packages MUST be supported by implementations of this specification, though this specification does not prevent additional API classes from being supported. Implementations are allowed to pull in a more recent patch version of the API classes.

### Tracing APIs

> - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-api/{otel-java-version}/io/opentelemetry/api/package-summary.html[io.opentelemetry.api](except `GlobalOpenTelemetry`)

Reviewer: ```suggestion
- https://www.javadoc.io/static/io.opentelemetry/opentelemetry-api/{otel-java-version}/io/opentelemetry/api/package-summary.html[io.opentelemetry.api] (except `GlobalOpenTelemetry`)
```

### Context API

- https://www.javadoc.io/static/io.opentelemetry/opentelemetry-context/1.28.0/io/opentelemetry/context/package-summary.html[io.opentelemetry.context]
- https://www.javadoc.io/static/io.opentelemetry/opentelemetry-context/1.28.0/io/opentelemetry/context/propagation/package-summary.html[io.opentelemetry.context.propagation]

### Resource SDK

- https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-common/1.28.0/io/opentelemetry/sdk/resources/package-summary.html[io.opentelemetry.sdk.resources]

### Autoconfigure SPI

This is the programmatic interface that allows users to register extensions when using the SDK Autoconfigure Extension (which we use for configuration). 

- https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-extension-autoconfigure-spi/1.28.0/io/opentelemetry/sdk/autoconfigure/spi/package-summary.html[io.opentelemetry.sdk.autoconfigure.spi]
- https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-extension-autoconfigure-spi/1.28.0/io/opentelemetry/sdk/autoconfigure/spi/traces/package-summary.html[io.opentelemetry.sdk.autoconfigure.spi.traces]

> The above packages have dependencies on the following packages:
> 
>   - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-trace/1.28.0/io/opentelemetry/sdk/trace/package-summary.html[io.opentelemetry.sdk.trace]
>   - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-trace/1.28.0/io/opentelemetry/sdk/trace/data/package-summary.html[io.opentelemetry.sdk.trace.data]
>   - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-trace/1.28.0/io/opentelemetry/sdk/trace/export/package-summary.html[io.opentelemetry.sdk.trace.export]
>   - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-trace/1.28.0/io/opentelemetry/sdk/trace/samplers/package-summary.html[io.opentelemetry.sdk.trace.samplers]
>   - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-common/1.28.0/io/opentelemetry/sdk/common/package-summary.html[io.opentelemetry.sdk.common]

Reviewer: ```suggestion
The above packages have dependencies on the following packages which must be supported to the extent that they are required by the Autoconfigure SPI classes:

  - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-trace/1.28.0/io/opentelemetry/sdk/trace/package-summary.html[io.opentelemetry.sdk.trace]
  - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-trace/1.28.0/io/opentelemetry/sdk/trace/data/package-summary.html[io.opentelemetry.sdk.trace.data]
  - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-trace/1.28.0/io/opentelemetry/sdk/trace/export/package-summary.html[io.opentelemetry.sdk.trace.export]
  - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-trace/1.28.0/io/opentelemetry/sdk/trace/samplers/package-summary.html[io.opentelemetry.sdk.trace.samplers]
  - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-sdk-common/1.28.0/io/opentelemetry/sdk/common/package-summary.html[io.opentelemetry.sdk.common]
```

### `WithSpan` annotation

> - https://www.javadoc.io/doc/io.opentelemetry.instrumentation/opentelemetry-instrumentation-annotations/latest/io/opentelemetry/instrumentation/annotations/package-summary.html[io.opentelemetry.instrumentation.annotations]

Reviewer: ```suggestion
- https://www.javadoc.io/doc/io.opentelemetry.instrumentation/opentelemetry-instrumentation-annotations/latest/io/opentelemetry/instrumentation/annotations/package-summary.html[io.opentelemetry.instrumentation.annotations] (`WithSpan` and `SpanAttribute` only)
```

> ==== `WithSpan` annotation

Reviewer: Had one thought, should this heading be "Annotations" to fit better with the other headings?

Reviewer: Sorry, I wasn't clear, I meant the whole heading should be

```
==== Annotations
```

### Semantic Conventions

> [NOTE]
> ====
> These packages are not stable, which might be subject to changes.
> 
> - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-semconv/1.28.0-alpha/io/opentelemetry/semconv/trace/attributes/package-summary.html[io.opentelemetry.semconv.trace.attributes]
> - https://www.javadoc.io/static/io.opentelemetry/opentelemetry-semconv/1.28.0-alpha/io/opentelemetry/semconv/resource/attributes/package-summary.html[io.opentelemetry.semconv.resource.attributes]
> ====

Reviewer: ```suggestion
[NOTE]
====
These packages are not stable and may be subject to breaking changes in future releases.
====

- https://www.javadoc.io/static/io.opentelemetry/opentelemetry-semconv/1.28.0-alpha/io/opentelemetry/semconv/trace/attributes/package-summary.html[io.opentelemetry.semconv.trace.attributes]
- https://www.javadoc.io/static/io.opentelemetry/opentelemetry-semconv/1.28.0-alpha/io/opentelemetry/semconv/resource/attributes/package-summary.html[io.opentelemetry.semconv.resource.attributes]
```
